api/docs/scatter_gather_emulation.dox

```diff
@@ -120,16 +239,23 @@ num_accesses = vector_size / element_size
 for i = 0, 1, ..., (num_accesses-1), do
   extract mask for the ith access from mask reg or mask vector
   if mask is set, then
-    extract ith element of index vector
-    compute address = base + ith index element
+    if index is vector, then
+        extract ith element of index vector
+        compute address = base + ith index element
+    else // base is vector
+        extract ith element of base vector
+        compute address = ith base element + index
```

Comment:
The index is multiplied by vector size too?

Reply:
The index isn't scaled by the vector length. Some instructions are unscaled, some scale the index by the element size.